### DISCUSSION TITLE
Event handling in Web Worker

### DISCUSSION THREAD

**Discussion by breadam** [pixijs/pixijs#8954] [Q&A]
Hey there, 

I'm using PIXI in a Web Worker and I am looking for ways to manually trigger event handler. 

Right now, I have an onclick event listener on canvas and I am sending the event data directly to the worker. But I don't know how to trigger that event in PIXI. Is it possible to manually do "hitTest" with the current API? I mean where should I look in the API, could you point me in the right direction please. Thank you,

**Comment by breadam** [comment]
I asked it to OpenGPT and here is the conversation. I did not test the answer but that's impressive nonetheless:)

![Screenshot_20221221_100743](https://user-images.githubusercontent.com/3389551/210158720-1e3b0681-3adc-4af7-95fc-ae356faab970.png)
![Screenshot_20221221_100810](https://user-images.githubusercontent.com/3389551/210158723-0e1b0e36-e4cd-44a4-8dfb-f9a5dc6d0396.png)
![Screenshot_20221221_100828](https://user-images.githubusercontent.com/3389551/210158725-ab7cb25f-04d6-424f-9ec7-63f36989f1fa.png)
![Screenshot_20221221_100841](https://user-images.githubusercontent.com/3389551/210158726-2e4181d3-f999-4aa5-bb3d-40bc32170688.png)

**Comment by SuperSodaSea** [comment]
Currently `@pixi/events`' code is coupled with some DOM APIs (for example in [`EventSystem.addEvents`](https://github.com/pixijs/pixijs/blob/d13ade4a05c3d695277d818c514467d016fcd53c/packages/events/src/EventSystem.ts#L353)), so it only works in browser main threads, and not supported in worker threads. Maybe we can decouple it later. You can make a [feature request](https://github.com/pixijs/pixijs/issues/new?assignees=&labels=Feature+Request&template=feature_request.yml&title=Feature+Request%3A+Support%20`@pixi/events`%20in%20Web%20Workers) for it, or if you want to work on this yourself, PR is welcomed.
